bftengine/include/bftengine/ReplicaConfig.hpp

```diff
@@ -35,6 +35,22 @@ namespace bftEngine {
 
 enum BatchingPolicy { BATCH_SELF_ADJUSTED, BATCH_BY_REQ_SIZE, BATCH_BY_REQ_NUM, BATCH_ADAPTIVE };
 
+enum class ExecutionEngine : char { DAML = 0, ETH = 1, TEE = 2, PERF = 3 };
+
+constexpr const char* execution_engine_text_identifier(ExecutionEngine executionEngine) noexcept {
+  switch (executionEngine) {
+    case ExecutionEngine::DAML:
+      return "DAML";
+    case ExecutionEngine::ETH:
+      return "ETH";
+    case ExecutionEngine::TEE:
+      return "TEE";
+    case ExecutionEngine::PERF:
+      return "PERF";
+  }
+  return "";
+}
```

Comment:
PS: can the execution engine specific part move to the closed source?

Reply:
@teoparvanov , I think you are right.

I think that this solution should be more abstract to hide the reason "why we are overriding" values. That means, when reading the code hear, no one is able to know we are overriding due to different conract layers/exec engine. Tomorrow we decide to override values due to othher reasons, it can be due to different scale of network - even with the same execution engine.

We need to be able to support overriding for any reasons, and usually hide the reason for that overriding at the code level - and make it clear on the file configuration (which is parto f vmwathena).

We won't need a huge change to be done here, just some more logical abstraction and renaming + moving some logic to vmwathena. 
@StefanPalashev, at this stage I don't want to start reviewing, lets meet tomorrow to discuss some additional changes.